recordval.go

```diff
@@ -2,12 +2,10 @@ package zed
 
```

Comment:
Maybe  move all of this to value.go and remove this file?

Reply:
From PR comment:
```
In a subsequent PR, we will merge recordval.go into value.go and delete
the former. For now, the diffs are easier to review on this commit.
```

Reply:
My bad!